subprojects/dependency-management/src/main/java/org/gradle/api/internal/notations/DependencyNotationParser.java

```diff
@@ -19,51 +19,123 @@
+            new NotationConverterToNotationParserAdapter<>(filesNotationConverter),
+            new NotationConverterToNotationParserAdapter<>(projectNotationConverter)
+        );
+    }
+
+    private final NotationParser<Object, Dependency> notationParser;
+    private final NotationParser<String, ? extends ExternalModuleDependency> stringNotationParser;
+    private final NotationParser<MinimalExternalModuleDependency, ? extends MinimalExternalModuleDependency> minimalExternalModuleDependencyNotationParser;
+    private final NotationParser<Map<String, ?>, ? extends ExternalModuleDependency> mapNotationParser;
+    private final NotationParser<FileCollection, ? extends FileCollectionDependency> fileCollectionNotationParser;
+    private final NotationParser<Project, ? extends ProjectDependency> projectNotationParser;
+
+    private DependencyNotationParser(NotationParser<Object, Dependency> notationParser,
+                                     NotationParser<String, ? extends ExternalModuleDependency> stringNotationParser,
+                                     NotationParser<MinimalExternalModuleDependency, ? extends MinimalExternalModuleDependency> minimalExternalModuleDependencyNotationParser,
+                                     NotationParser<Map<String, ?>, ? extends ExternalModuleDependency> mapNotationParser,
+                                     NotationParser<FileCollection, ? extends FileCollectionDependency> fileCollectionNotationParser,
+                                     NotationParser<Project, ? extends ProjectDependency> projectNotationParser
+    ) {
+        this.notationParser = notationParser;
+        this.stringNotationParser = stringNotationParser;
+        this.minimalExternalModuleDependencyNotationParser = minimalExternalModuleDependencyNotationParser;
+        this.mapNotationParser = mapNotationParser;
+        this.fileCollectionNotationParser = fileCollectionNotationParser;
+        this.projectNotationParser = projectNotationParser;
+    }
+
+    public NotationParser<Object, Dependency> getNotationParser() {
+        return notationParser;
+    }
+
+    public NotationParser<String, ? extends ExternalModuleDependency> getStringNotationParser() {
+        return stringNotationParser;
+    }
+
+    public NotationParser<MinimalExternalModuleDependency, ? extends MinimalExternalModuleDependency> getMinimalExternalModuleDependencyNotationParser() {
+        return minimalExternalModuleDependencyNotationParser;
+    }
+
+    public NotationParser<Map<String, ?>, ? extends ExternalModuleDependency> getMapNotationParser() {
+        return mapNotationParser;
+    }
+
+    public NotationParser<FileCollection, ? extends FileCollectionDependency> getFileCollectionNotationParser() {
+        return fileCollectionNotationParser;
+    }
+
+    public NotationParser<Project, ? extends ProjectDependency> getProjectNotationParser() {
+        return projectNotationParser;
     }
```

Comment:
Can you add test coverage exercising these parsers if they stay?

Reply:
What exactly do you want covered? The parsers are the same ones already used by the DependencyHandler, so their functionality is well-covered. The only thing I can think of is the getters, but is that worth writing tests for?

Reply:
Hmm, this might not be as bad as I originally thought. I was concerned about overlapping parsers that only behaved correctly when combined. It looks like we don't have any overlapping parsers, so this shouldn't be a problem.